docs/software-architecture-and-design.md

```diff
@@ -472,6 +475,30 @@ The most important thing to know about editing the [Visualizer](#visualizer) in
 
 More information about defining custom widgets in [QtCreator](https://doc.qt.io/qtcreator/creator-using-qt-designer.html) can be found [here](https://doc.qt.io/qt-5/designer-using-custom-widgets.html).
 
+## Qt Best Practices
+Although this will focus on Qt-specific examples, these principles generally apply to all GUI design and implementation. They are really just Software Engineering principles applied to GUIs.
+
+### Build A Hierarchy
+Qt is designed to handle hierarchy. It has [an extensive and robust system for maintaining Object Trees and parent/child relationships between components](https://doc.qt.io/qt-5/objecttrees.html). Much like regular code, GUIs should be created in a logical hierarchy to make organization and re-use easier.
+
+Do **not** lay out all your widgets in a single layer like this:
+![Bad Qt Widget Layout](images/qt_layout_bad_example.png)
+
+Make sure to use layouts, and group widgets in a logical way. For example, several widgets that work together to collect a user's mailing address should be grouped. This group may then be part of a larger group of widgets in a popup dialog asking for billing information.
+
+Here is a good example of laying out widgets in a hierarchy:
+![Good Qt Widget Layout](images/qt_layout_good_example.png)
```

Comment:
I feel like putting a good example of the hierarchy first before the bad example makes sense.
As a reader, I would feel better if I had a sense of what I should be doing before being presented with "Don't do this."

Reply:
Re-ordered. Lmk if that's better

Reply:
Improves readability :+1: